pymeasure/instruments/anaheimautomation/dpseriesmotorcontroller.py

```diff
@@ -0,0 +1,377 @@
+            pos = self.ask("VEP")
+        else:
+            pos = self.ask("VZ")    
+        return int(pos)
+    
+    @step_position.setter
+    def step_position(self, pos):
+        """ Sends command to the motor controller to move to the desired step position. Note that in DP series
+            controller instrument manuals, this property corresponds to the 'absolute position' command. In this driver,
+            the `absolute_position` property implements a conversion between steps to absolute units whereas the DP
+            series manuals refer to `absolute position` as measured in steps.
+        """
+        step_pos = int(pos)
+        if -8388607 < step_pos < 8388607:
+            self.write("P%i" % step_pos)
+            self.go()
+        else:
+            raise ValueError("Provided step position out of valid range!")
+    
+    @property
+    def absolute_position(self):
+        """
+        Return the absolute position of the motor in units determined by the steps_to_position() method.
+        """
+        step_pos = self.step_position
+        return self.steps_to_absolute(step_pos)
+    
+    @absolute_position.setter
+    def absolute_position(self, abs_pos):
+        """ 
+        Move the motor to the absolute position provided. Note that the absolute_to_steps()
+        method must be implemented in a subclass for this to work.
+        
+        :param abs_pos: absolute position in units determined by the absolute_to_steps() method.
+        """
+        steps_pos = self.absolute_to_steps(abs_pos)
+        self.step_position = steps_pos
+    
+    def absolute_to_steps(self, pos):
+        """ Convert an absolute position to a number of steps to move. This must be implemented in subclasses.
+        
+        :param pos: Absolute position in the units determined by the subclassed position_to_steps() method.
+        """
+        raise NotImplementedError("absolute_to_steps() must be implemented in subclasses!")   
+    
+    def steps_to_absolute(self, steps):
+        """ Convert a position measured in steps to an absolute position.
+        
+        :param steps: Position in steps to be converted to an absolute position.
+        """
+        raise NotImplementedError("steps_to_position() must be implemented in subclasses!")
+    
+    def reset_position(self):
+        """ Reset the position as counted by the motor controller and an externally connected encoder to 0.
+        """
+        # reset encoder recorded position #
+        self.write("ET")
+        # reset motor recorded position #
+        self.write("Z0")
+
+    def stop(self):
+        """Method that stops all motion on the motor controller."""
+        self.write(".")
+
+    def go(self):
+        """Method that sends the G command to the controller to tell the controller to turn the 
+           motor the number of steps previously set with the steps property."""
+        self.write("G")
+
+    def step(self, steps, direction):
```

Comment:
Might be worth it to name this one `step_distance` or `step_for()` to increase clarity.